keyboards/clueboard/2x1800/2019/2019.c

```diff
@@ -0,0 +1,196 @@
+    setPinOutput(B5);
+    setPinOutput(B6);
+
+    // Set our Tilt Sensor pins as input
+    setPinInputHigh(SHAKE_PIN_A);
+    setPinInputHigh(SHAKE_PIN_B);
+
+    // Run the keymap level init
+    matrix_init_user();
+}
+
+#ifdef DRAWING_ENABLE
+bool drawing_mode = false;
+bool btn1_pressed = false;
+bool btn2_pressed = false;
+bool btn3_pressed = false;
+bool btn4_pressed = false;
+
+void check_encoder_buttons(void) {
+    if (btn1_pressed && btn2_pressed && btn3_pressed && btn4_pressed) {
+        // All 4 buttons pressed, toggle drawing mode
+	if (drawing_mode) {
+            dprintf("Turning drawing mode off.\n");
+            drawing_mode = false;
+            writePinLow(D6);
+	    unregister_code(KC_BTN1);
+	} else {
+            dprintf("Turning drawing mode on.\n");
+            drawing_mode = true;
+            writePinHigh(D6);
+	    register_code(KC_BTN1);
+	}
+    }
+}
+#endif
+
+#ifdef SHAKE_ENABLE
+uint8_t tilt_state = 0x11;
+uint8_t detected_shakes = 0;
+static uint16_t shake_timer;
+#endif
+
+void matrix_scan_kb(void) {
+#ifdef SHAKE_ENABLE
+    // Read the current state of the tilt sensor. It is physically
+    // impossible for both pins to register a low state at the same time.
+    uint8_t tilt_read = (readPin(SHAKE_PIN_A) << 4) | readPin(SHAKE_PIN_B);
+
+    // Check to see if the tilt sensor has changed state since our last read
+    if (tilt_state != tilt_read) {
+        shake_timer = timer_read();
+        detected_shakes++;
+        tilt_state = tilt_read;
+    }
+
+    if ((detected_shakes > 0) && (timer_elapsed(shake_timer) > SHAKE_TIMEOUT)) {
+        if (detected_shakes > SHAKE_COUNT) {
+            dprintf("Shake triggered! We detected %d shakes.\n", detected_shakes);
+            tap_code16(SHAKE_KEY);
+        } else {
+            dprintf("Shake not triggered! We detected %d shakes.\n", detected_shakes);
+        }
+        detected_shakes = 0;
+    }
+#endif
+
+    matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+#ifdef DRAWING_ENABLE
+    if (keycode == ENC_BTN1) {
+        if (record->event.pressed) {
+            btn1_pressed = true;
+	    register_code(KC_BTN1);
+	} else {
+            btn1_pressed = false;
+	    unregister_code(KC_BTN1);
+	}
+    }
+    if (keycode == ENC_BTN2) {
+        if (record->event.pressed) {
+            btn2_pressed = true;
+	    register_code(KC_BTN2);
+	} else {
+            btn2_pressed = false;
+	    unregister_code(KC_BTN2);
+	}
+    }
+    if (keycode == ENC_BTN3) {
+        if (record->event.pressed) {
+            btn3_pressed = true;
+	    register_code(KC_BTN3);
+	} else {
+            btn3_pressed = false;
+	    unregister_code(KC_BTN3);
+	}
+    }
+    if (keycode == ENC_BTN4) {
+        if (record->event.pressed) {
+            btn4_pressed = true;
+	    register_code(KC_BTN4);
+	} else {
+            btn4_pressed = false;
+	    unregister_code(KC_BTN4);
+	}
+    }
+#endif
+
+    check_encoder_buttons();
```

Comment:
I'm pretty sure this only needs to be called if drawing mode is enabled.

In fact, I don't think the function exists otherwise. 

```suggestion

    check_encoder_buttons();
#endif

```